### ISSUE TITLE
Error: actualValue.toLowerCase is not a function

### ISSUE THREAD

**Issue by iamEAP** [run-crank/cog-salesforce#55]
### Repro Steps
1. Install `v0.4.0` of the Salesforce cog (`crank cog:install automatoninc/salesforce:0.4.0 --force --ignore-auth`)
2. Run the **Check a field on a Salesforce Account** step (`crank cog:step automatoninc/salesforce --step=AccountFieldEquals` with...
  - **ID Field** set to `Name`
  - **ID Field Value** set to `University of Arizona`
  - **Check Field** set to `BillingAddress`
  - **Operator** set to `Contain`
  - **Expected Value** set to `85721`

### Expected
The step should pass, since the value of the field contains the above sequence of characters.

### Actual
The step results in an error because `.toLowerCase()` is attempted to be used on a number.

![Screen Shot 2019-12-31 at 12 47 51 PM](https://user-images.githubusercontent.com/3496491/71633897-2c9c8780-2bcc-11ea-9883-b58ba1f58f5d.png)

**Comment by iamEAP** [comment]
I had originally thought this was a bug in the way the `util.compare` method was handling values that were purely numeric, but I found the same error even when it was string-ish (e.g. `AZ 85721`).

I think it's actually specifically an issue with the `BillingAddress` field, which appears to come back as an object from the API:

```
{ city: 'Tucson',
  country: null,
  geocodeAccuracy: null,
  latitude: null,
  longitude: null,
  postalCode: null,
  state: 'AZ',
...
```

...So I've moved the issue over to the Salesforce Cog repo.

A simple/generic fix across all object types would be to do a type check in the base step compare method and, if it the actualValue was non-scalar, make it scalar (e.g. `JSON.stringify()`).  ...Or we could choose to handle specific, known core fields (but this could be tedious).
